defender/downloaded/Autotask_Forta_Explorer_Monitor.js

```diff
@@ -0,0 +1,474 @@
+            name
+          }
+        }
+        nextPageValues {
+          blocknumber
+          id
+        }
+        currentPageValues {
+          blocknumber
+          id
+        }
+      }
+    }`,
+    variables: {
+      getListInput: {
+        severity: [],
+        startDate: (lastUpdateTimestamp + 1).toString(),
+        endDate: currentTimestamp.toString(),
+        txHash: '',
+        text: '',
+        muted: [],
+        limit: 0,
+        sort: 'desc',
+        agents: [botId],
+        addresses: [],
+        project: '',
+      },
+    },
+  };
+  return graphqlQuery;
+}
+
+// this query gathers information used to populate fields on the page for the Bot
+// specifically, this contains data such as the Bot ID, Image, Last Updated, Enabled, etc.
+function createAgentInformationQuery(id) {
+  const graphqlQuery = {
+    operationName: 'Retrieve',
+    query: `query Retrieve($getAgentInput: AgentInformation) {
+      getAgentInformation(input: $getAgentInput) {
+        id
+        name
+        developer
+        chainIds
+        projects
+        created_at
+        updated_at
+        description
+        version
+        repository
+        enabled
+        image
+        manifest_ipfs
+        doc_ipfs
+      }
+    }`,
+    variables: {
+      getAgentInput: {
+        id,
+      },
+    },
+  };
+  return graphqlQuery;
+}
+
+// timeFrame values: 'hour', 'day', 'week', 'month'
+function createMetricsQuery(agentId, timeFrame) {
+  const graphqlQuery = {
+    query: `query ($getAgentMetricsInput: GetAgentMetricsInput) {
+      getAgentMetrics(input: $getAgentMetricsInput) {
+        metrics {
+          key
+          scanners {
+            key
+            interval {
+              key
+              sum
+              max
+            }
+          }
+        }
+      }
+    }`,
+    variables: {
+      getAgentMetricsInput: {
+        agentId,
+        timeFrame,
+      },
+    },
+  };
+  return graphqlQuery;
+}
+
+async function postQuery(graphqlQuery) {
+  const headers = {
+    'content-type': 'application/json',
+  };
+
+  // perform the POST request
+  const response = await axios({
+    url: fortaExplorerApiEndpoint,
+    method: 'post',
+    headers,
+    data: graphqlQuery,
+  });
+
+  return response;
+}
+
+function calculateTimeFrame(currentTimestamp, lastUpdateTimestamp) {
+  // if this is the first time the Autotask has executed, gather data for the last month
+  // on subsequent runs, set the time frame based on the previous timestamp and the current
+  // timestamp
+  const millisecondsPerHour = 60 * 60 * 1000;
+  const millisecondsPerDay = millisecondsPerHour * 24;
+  const millisecondsPerWeek = millisecondsPerDay * 7;
+
+  // set the time frame based on the previous timestamp and the current timestamp
+  const deltaTimestamp = currentTimestamp - lastUpdateTimestamp;
+  if (deltaTimestamp <= millisecondsPerHour) {
+    return 'hour';
+  } else if (deltaTimestamp <= millisecondsPerDay) {
```

Comment:
```suggestion
  } if (deltaTimestamp <= millisecondsPerDay) {
```
Else isn't needed since it it will return before it hits the next if statement. https://eslint.org/docs/rules/no-else-return